bzt/modules/jmeter.py

```diff
@@ -83,6 +83,161 @@ def __init__(self):
         self.stderr_file = None
         self.tool = None
 
+    def get_load(self):
+        """
+        Helper method to read load specification
+        """
+        prov_type = self.engine.config.get(Provisioning.PROV)
+
+        ensure_is_dict(self.execution, ScenarioExecutor.THRPT, prov_type)
+        throughput = self.execution[ScenarioExecutor.THRPT].get(prov_type, 0)
+
+        ensure_is_dict(self.execution, ScenarioExecutor.CONCURR, prov_type)
+        concurrency = self.execution[ScenarioExecutor.CONCURR].get(prov_type, 0)
+
+        iterations = self.execution.get("iterations", None)
+
+        steps = self.execution.get(ScenarioExecutor.STEPS, None)
+
+        hold = self.execution.get(ScenarioExecutor.HOLD_FOR, 0)
+        try:
+            hold = dehumanize_time(hold)
+        except:
+            hold = 0
+
+        ramp_up = self.execution.get(ScenarioExecutor.RAMP_UP, None)
+        if ramp_up is None:
+            duration = hold
+        else:
+            try:
+                ramp_up = dehumanize_time(ramp_up)
+            except:
+                ramp_up = 0
+
+            duration = hold + ramp_up
+
+        msg = ''
+        if not isinstance(concurrency, numeric_types + (type(None),)):
+            msg += "Invalid concurrency value[%s]: %s " % (type(concurrency).__name__, concurrency)
+        if not isinstance(throughput, numeric_types + (type(None),)):
+            msg += "Invalid throughput value[%s]: %s " % (type(throughput).__name__, throughput)
+        if not isinstance(steps, numeric_types + (type(None),)):
+            msg += "Invalid throughput value[%s]: %s " % (type(steps).__name__, steps)
+        if not isinstance(iterations, numeric_types + (type(None),)):
+            msg += "Invalid throughput value[%s]: %s " % (type(iterations).__name__, iterations)
+
+        if msg:
+            self.log.warning(msg)
+
+        try:
+            throughput = float(throughput)
+        except:
+            throughput = 0
+
+        try:
+            concurrency = int(concurrency)
+        except:
+            concurrency = 1
+
+        try:
+            iterations = int(iterations)
+        except:
+            iterations = 0
+
+        try:
+            steps = int(steps)
+        except:
+            steps = 0
+
+        if duration and not iterations:
+            iterations = 0  # which means infinite
+
+        res = namedtuple("LoadSpec",
```

Comment:
If this type is common with engine.py, it should be refactored out for consistency.